src/shared/utils/utils.ts

```diff
@@ -89,3 +89,20 @@ export const isSameEntry = (
   prev?.primaryAccession !== undefined
     ? prev.primaryAccession === next.primaryAccession
     : prev.id === next.id;
+
+export const deepFindAllByKey = (
+  o: Record<string, unknown> | Record<string, unknown>[],
+  predicateKey: string,
+  accum: string[] = []
+) => {
+  if (typeof o === 'object') {
+    Object?.entries(o).forEach(([key, value]) => {
+      if (key === predicateKey) {
+        accum.push(value as string);
+      } else {
+        deepFindAllByKey(value as Record<string, unknown>, predicateKey, accum);
+      }
+    });
+  }
+  return accum;
+};
```

Comment:
If we have to do recursivity I'd rather avoid passing an accumulator around and mutate it, and instead would go for a generator.
See suggestion, feel free to use as is or modify.
```suggestion
export function* deepFindAllByKey<T = string>(
  input: unknown,
  predicateKey: string
): Generator<T, void, never> {
  if (Array.isArray(input)) {
    for (const item of input) {
      yield* deepFindAllByKey<T>(item, predicateKey);
    }
  } else if (input && typeof input === 'object') {
    for (const [key, value] of Object.entries(input)) {
      if (key === predicateKey) {
        yield value;
      } else {
        yield* deepFindAllByKey(value, predicateKey);
      }
    }
  }
}
```
Wasn't sure about the types, so I changed it a bit. Now accepting anything, and returning the correct type without having to cast inside the function. The expected type can optionally be set by the user on calling the generator, but assuming it is a string by default.

Note that for its use you'd need to wrap the current places where the function is called in this way: `Array.from(deepFindAllByKey(/* */))`.

Although... If the use case for this function is to only be used in the context of unique strings, then I'd keep the current approach, and directly use a `Set` as an accumulator, not an array, so that you don't have to wrap it with `lodash`'s `uniq`